code/modules/mob/living/carbon/human/species_types/shadowpeople.dm

```diff
@@ -7,17 +7,89 @@
 	ignored_by = list(/mob/living/simple_animal/hostile/faithless)
 	meat = /obj/item/reagent_containers/food/snacks/meat/slab/human/mutant/shadow
 	species_traits = list(NOBREATH,NOBLOOD,RADIMMUNE,VIRUSIMMUNE)
+
 	dangerous_existence = 1
 	mutanteyes = /obj/item/organ/eyes/night_vision
 
 
 /datum/species/shadow/spec_life(mob/living/carbon/human/H)
-	var/light_amount = 0
-	if(isturf(H.loc))
-		var/turf/T = H.loc
-		light_amount = T.get_lumcount()
+	var/turf/T = H.loc
+	if(istype(T))
+		var/light_amount = T.get_lumcount()
 
-		if(light_amount > 0.2) //if there's enough light, start dying
+		if(light_amount > SHADOW_SPECIES_LIGHT_THRESHOLD) //if there's enough light, start dying
 			H.take_overall_damage(1,1)
-		else if (light_amount < 0.2) //heal in the dark
+		else if (light_amount < SHADOW_SPECIES_LIGHT_THRESHOLD) //heal in the dark
 			H.heal_overall_damage(1,1)
+
+
+/datum/species/shadow/nightmare
+	name = "Nightmare"
+	id = "nightmare"
+	limbs_id = "shadow"
+	burnmod = 1.5
+	blacklisted = TRUE
+	no_equip = list(slot_wear_mask, slot_wear_suit, slot_gloves, slot_shoes, slot_w_uniform, slot_s_store)
+	species_traits = list(NOBREATH,RESISTCOLD,RESISTPRESSURE,NOGUNS,NOBLOOD,RADIMMUNE,VIRUSIMMUNE,PIERCEIMMUNE,NODISMEMBER,NO_UNDERWEAR,NOHUNGER)
+	mutanteyes = /obj/item/organ/eyes/night_vision/nightmare
+
+/datum/species/shadow/nightmare/on_species_gain(mob/living/carbon/C, datum/species/old_species)
+	. = ..()
+	var/obj/effect/proc_holder/spell/targeted/shadowwalk/SW = new
+	C.AddSpell(SW)
+	var/obj/item/light_eater/blade = new
+	C.put_in_hands(blade)
+
+
+/datum/species/shadow/nightmare/bullet_act(obj/item/projectile/P, mob/living/carbon/human/H)
+	var/turf/T = H.loc
+	if(istype(T))
+		var/light_amount = T.get_lumcount()
+		if(light_amount < SHADOW_SPECIES_LIGHT_THRESHOLD)
+			H.visible_message("<span class='danger'>[H] dances in the shadows, evading [P]!</span>")
+			playsound(T, "bullet_miss", 75, 1)
+			return -1
+	return 0
+
+/obj/item/light_eater
+	name = "light eater"
+	icon_state = "arm_blade"
+	item_state = "arm_blade"
+	force = 25
+	armour_penetration = 35
+	lefthand_file = 'icons/mob/inhands/antag/changeling_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/antag/changeling_righthand.dmi'
+	flags_1 = ABSTRACT_1 | NODROP_1 | DROPDEL_1
+	w_class = WEIGHT_CLASS_HUGE
+	sharpness = IS_SHARP
+
+/obj/item/light_eater/afterattack(atom/movable/AM, mob/user, proximity)
+	if(!proximity)
+		return
+	if(isopenturf(AM)) //So you can actually melee with it
+		return
+	if(isliving(AM))
+		if(iscyborg(AM))
+			var/mob/living/silicon/robot/borg = AM

```

Comment:
Just make this generic function , extinguish_light or similar instead of all these istypes.

Reply:
Is an atom level proc for one weapon better than a handful of istypes?